### ISSUE TITLE
only-arrow-functions: allow for async function

### ISSUE THREAD

**Issue by retorquere** [palantir/tslint#3319]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
transactional(async function() {
        // ...
        await conn.queryAsync()
       // ...
})
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
      "ban-types": {
        "severity": "warning",
        "options": [
          ["Object", "Avoid using the `Object` type. Did you mean `object`?"],
          ["Function", "Avoid using the `Function` type. Prefer a specific function type, like `() => void`."],
          ["Boolean", "Avoid using the `Boolean` type. Did you mean `boolean`?"],
          ["Number", "Avoid using the `Number` type. Did you mean `number`?"],
          ["String", "Avoid using the `String` type. Did you mean `string`?"],
          ["Symbol", "Avoid using the `Symbol` type.  Did you mean `symbol`?"]
        ]
      },
      "max-classes-per-file": [true, 3],
      "new-parens": false,
      "no-var-requires": false,
      "quotemark": [true, "single", "avoid-escape", "avoid-template"],
      "max-line-length": [true, 240],
      "prefer-const": [true, {"destructuring": "all"}],
      "member-ordering": "statics-first",
      "no-magic-numbers": true,
      "prefer-for-of": true,
      "no-duplicate-variable": true,
      "no-string-throw": true,
      "no-unused-variable": [true, {"ignore-pattern": "^_"}],
      "no-use-before-declare": true,
      "no-var-keyword": true,
      "prefer-object-spread": true,
      "triple-equals": [true, "allow-undefined-check", "allow-null-check"],
      "typeof-compare": true,
      "indent": [true, "spaces", 2],
      "linebreak-style": [true, "LF"],
      "eofline": true,
      "trailing-comma": [true, {"singleline": "never", "multiline": {"objects": "always", "arrays": "always", "functions": "never", "typeLiterals": "ignore"}}],
      "arrow-parens": [true, "ban-single-arg-parens"],
      "arrow-return-shorthand": true,
      "binary-expression-operand-order": true,
      "class-name": true,
      "no-irregular-whitespace": true,
      "no-trailing-whitespace": [true, "ignore-jsdoc", "ignore-template-strings"],
      "no-unnecessary-callback-wrapper": true,
      "object-literal-key-quotes": [true, "as-needed"],
      "object-literal-shorthand": true,
      "one-line": [true, "check-catch", "check-finally", "check-else", "check-open-brace"],
      "prefer-template": [true, "allow-single-concat"],
      "semicolon": [true, "never"],
      "space-before-function-paren": [true, "never"],
      "variable-name": [true, "ban-keywords", "check-format", "allow-leading-underscore", "allow-snake-case" ],
      "curly": [true, "ignore-same-line"],
      "only-arrow-functions": [true, "allow-declarations", "allow-named-functions"],
      "radix": false,
      "object-literal-sort-keys": false,
      "no-empty": [true, "allow-empty-catch"],
      "no-conditional-assignment": false,
      "one-variable-per-declaration": false
    },
    "rulesDirectory": [],
    "doc": "https://palantir.github.io/tslint/rules/"
}
```

#### Actual behavior

I'm getting the error `only-arrow-functions  non-arrow functions are forbidden`

#### Expected behavior

A way to configure the rule to allow unnamed non-arrow functions when they call `await` (which is not supported in arrow functions)

**Comment by ajafff** [comment]
Javascript (and typescript) allow async arrow functions and they can use `await` inside their body.

**Comment by retorquere** [comment]
Then I was misinformated -- apologies
